Data/Sequence/Internal.hs

```diff
@@ -4624,7 +4642,22 @@ unstableSortBy cmp (Seq xs) =
     maybe
         (Seq EmptyT)
         (execState (replicateA (size xs) (popMin cmp)))
-        (toPQ cmp (Seq xs))
+        (toPQ cmp (\(Elem x) -> PQueue x Nil) xs)
+
+-- | \( O(n \log n) \). `unstableSortOn` sorts the specified `Seq` by
+-- comparing the results of a key function applied to each element.
+-- `unstableSortOn f` is equivalent to `unstableSortBy (comparing f)`,
+-- but has the performance advantage of only evaluating f once for each
+-- element in the input list. This is called the
+-- decorate-sort-undecorate paradigm, or Schwartzian transform.
+unstableSortOn :: Ord b => (a -> b) -> Seq a -> Seq a
+unstableSortOn f (Seq xs) =
+    maybe
+       (Seq EmptyT)
+       (execState (replicateA (size xs) (fmap (\(_ :*: y) -> y) (popMin compareFst))))
```

Comment:
Any reason you used y in the lambda here but x in the lambda for the stable version?

Reply:
Nope! I'll change it to be consistent.